scripts/configurator/parser/cmake_row.py

```diff
@@ -49,9 +49,9 @@ def __init__(self, raw_row: str, row_number: int):
       row  = raw_row.strip()
       matches: Optional[Match[str]]  = re.match(self.regex_expression, row)
       self.type = row_type.NOT_PARSABLE
-      self.val: str = ""
+      self.val: str | bool = ""
```

Comment:
I appreciate the effort here. That certainly is way better than before, but I still don't like the initialization to empty string. Empty scripts used as a marker for unset is a very bad **anti-pattern**: please-don't do that. Initialize to `None` and deal with that.

About the type, what I really wanted was `Any` because that's what `val` is (it could be anything) but.. if it's possible to specify a list of types instead (as you're doing), I guess that could work as well.

Reply:
I could create a composite type, if you feel like that would reflect better the idea we are trying to convey